crates/rattler_shell_helpers/src/lib.rs

```diff
@@ -0,0 +1,354 @@
+use std::{
+    ffi::OsStr,
+    fs,
+    path::{Path, PathBuf},
+};
+
+use indexmap::IndexMap;
+
+#[derive(Copy, Clone, Debug)]
+pub enum ShellType {
+    Bash,
+    Zsh,
+    Fish,
+    Powershell,
+    CmdExe,
+}
+
+impl ShellType {
+    /// Returns the file extension for this shell type.
+    fn suffix(&self) -> &'static OsStr {
+        match self {
+            ShellType::Bash => OsStr::new("sh"),
+            ShellType::Zsh => OsStr::new("zsh"),
+            ShellType::Fish => OsStr::new("fish"),
+            ShellType::Powershell => OsStr::new("ps1"),
+            ShellType::CmdExe => OsStr::new("bat"),
+        }
+    }
+}
+
+#[derive(Copy, Clone, Debug)]
+pub enum OperatingSystem {
+    /// The Windows operating system
+    Windows,
+
+    /// The UNIX family (Linux or macOS) operating systems
+    Unix,
+}
+
+/// A struct that holds values for the activation and deactivation
+/// process of an environment, e.g. activation scripts to execute or environment variables to set.
+pub struct Activator {
+    /// The path to the root of the conda environment
+    pub target_prefix: PathBuf,
+
+    /// The type of shell that is being activated
+    pub shell_type: ShellType,
+
+    /// Paths that need to be added to the PATH environment variable
+    pub paths: Vec<PathBuf>,
+
+    /// A list of scripts to run when activating the environment
+    pub activation_scripts: Vec<PathBuf>,
+
+    /// A list of scripts to run when deactivating the environment
+    pub deactivation_scripts: Vec<PathBuf>,
+
+    /// A list of environment variables to set when activating the environment
+    pub env_vars: IndexMap<String, String>,
```

Comment:
Are you using the  `IndexMap` so you can sort them :)?

Reply:
Nope, so that they stay sorted. It's not well specified and IMO the conda folks chose the wrong json format (it should've been an ordered list) ...

Reply:
I am using the `preserve_order` feature of serde, btw.